cosmic-core/server/src/main/java/com/cloud/api/query/QueryManagerImpl.java

```diff
@@ -3614,6 +3615,51 @@ private void buildAffinityGroupViewSearchCriteria(final SearchCriteria<AffinityG
         return whoHasThisIpList;
     }
 
+    public ListResponse<WhoHasThisIpResponse> listWhoHasThisMac(final ListWhoHasThisMacCmd cmd) {
```

Comment:
This method shares significant logic with `listWoHasThisIp`. They could be refactored to share that logic.